src/overlays/actors/ovl_kaleido_scope/z_kaleido_item.c

```diff
@@ -1,9 +1,813 @@
+                        } else {
+                            gSaveContext.equips.buttonItems[2] = gSaveContext.equips.buttonItems[3];
+                            gSaveContext.equips.cButtonSlots[1] = gSaveContext.equips.cButtonSlots[2];
+                            Interface_LoadItemIcon2(globalCtx, 2);
+                        }
+                    } else {
+                        gSaveContext.equips.buttonItems[2] = ITEM_NONE;
+                        gSaveContext.equips.cButtonSlots[1] = SLOT_NONE;
+                    }
+                }
+
+                if ((pauseCtx->equipTargetItem >= 0xBF) && (pauseCtx->equipTargetItem <= 0xC1)) {
+                    if ((gSaveContext.equips.buttonItems[3] == ITEM_BOW) ||
+                        ((gSaveContext.equips.buttonItems[3] >= ITEM_BOW_ARROW_FIRE) &&
+                         (gSaveContext.equips.buttonItems[3] <= ITEM_BOW_ARROW_LIGHT))) {
+                        pauseCtx->equipTargetItem -= 0xBF - ITEM_BOW_ARROW_FIRE;
+                        pauseCtx->equipTargetSlot = SLOT_BOW;
+                    }
+                } else if (pauseCtx->equipTargetItem == ITEM_BOW) {
+                    if ((gSaveContext.equips.buttonItems[1] >= ITEM_BOW_ARROW_FIRE) &&
+                        (gSaveContext.equips.buttonItems[1] <= ITEM_BOW_ARROW_LIGHT)) {
+                        gSaveContext.equips.buttonItems[1] = gSaveContext.equips.buttonItems[3];
+                        Interface_LoadItemIcon2(globalCtx, 1);
+                    } else if ((gSaveContext.equips.buttonItems[2] >= ITEM_BOW_ARROW_FIRE) &&
+                               (gSaveContext.equips.buttonItems[2] <= ITEM_BOW_ARROW_LIGHT)) {
+                        gSaveContext.equips.buttonItems[2] = gSaveContext.equips.buttonItems[3];
+                        Interface_LoadItemIcon2(globalCtx, 2);
+                    }
+                }
+
+                gSaveContext.equips.buttonItems[3] = pauseCtx->equipTargetItem;
+                gSaveContext.equips.cButtonSlots[2] = pauseCtx->equipTargetSlot;
+                Interface_LoadItemIcon1(globalCtx, 3);
+
+                osSyncPrintf("Ｃ右sl_item_no=%d (1)=%d (2)=%d (3)=%d\n", pauseCtx->equipTargetItem,
+                             gSaveContext.equips.buttonItems[1], gSaveContext.equips.buttonItems[2],
+                             gSaveContext.equips.buttonItems[3]);
+                osSyncPrintf("Ｃ右sl_number=%d (1)=%d (2)=%d (3)=%d\n", pauseCtx->equipTargetSlot,
+                             gSaveContext.equips.cButtonSlots[0], gSaveContext.equips.cButtonSlots[1],
+                             gSaveContext.equips.cButtonSlots[2]);
+            }
+
+            pauseCtx->unk_1E4 = 0;
+            sEquipMoveTimer = 10;
+            WREG(90) = 320;
+            WREG(87) = WREG(91);
+        }
+        return;
+    }
+
+    sEquipAnimTimer--;
+
+    if (sEquipAnimTimer == 0) {
+        pauseCtx->equipAnimAlpha = 255;
+    }
```

Comment:
Then wrap the stuff at the end (line 808) in an else to remove the `return` at line 805.

```suggestion
        }
        return;
    }

    sEquipAnimTimer--;

    if (sEquipAnimTimer == 0) {
        pauseCtx->equipAnimAlpha = 255;
    }
```
```suggestion
        }
    }
    else {
        sEquipAnimTimer--;

        if (sEquipAnimTimer == 0) {
            pauseCtx->equipAnimAlpha = 255;
        }
    }
```

Reply:
An else at the end here seems fine tho, so I'll change it as well